### PR TITLE
SITES-479: Preserve destination parameter.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Properly handles redirects for log in.

# Needed By (Date)
- End of June

# Urgency
- 1

# Steps to Test

1. Deploy this on https://sites576-dev.cardinalsites.stanford.edu/
2. Click the login link
3. Validate you end up on the right page.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by jbickar** [comment]
Works in testing on local Lando.

File: stanford_ssp.module [hunk 1]
```diff
@@ -144,17 +144,25 @@ function stanford_ssp_access_login_page() {
   // If the user is already logged in and a goto is set then do the redirect.
   if (isset($_GET['goto'])) {
     $goto = check_plain($_GET['goto']);
-    if (drupal_valid_path($goto)) {
```
Comment (jbickar) [inline stanford_ssp.module]: Why are you removing the check for `drupal_valid_path()` here?

Comment (sherakama) [reply stanford_ssp.module]: I guess I should have called that out. Good question and thanks for asking.

I removed the valid path option as redirects do not resolve as a valid path. So for example, if I wanted to go to /resources but /resources was behind webauth and redirected to /resources/overview I would be automagically logged in and sent to my user page as the redirect would not resolve as a valid path.

Comment (jbickar) [reply stanford_ssp.module]: Got it, GTK.

**Pull request merged**